### PR TITLE
refactor(@desktop/settings-ens-usernames): ens usernames module added and corresponding ui updated

### LINKED ISSUE #4394: refactor ens usernames frontend in base bc branch

### PR DESCRIPTION
**Pull request by saledjenic**
A complete logic used in these changes is more or less the same as it was in the "old" code, just moved to the new arch and updated a bit accordingly. What we need to do later is to move some parts from `ens` service to `wallet`/`transaction`/`eth` services.

Corresponding `status-lib` PR:
- https://github.com/status-im/status-lib/pull/168

Fixes #4394

### CODE REVIEW

File: src/app_service/service/ens/service.nim [hunk 1]
```diff
@@ -0,0 +1,249 @@
+import NimQml, Tables, sets, json, sequtils, strutils, strformat, chronicles
```
Comment (alaibe) [inline src/app_service/service/ens/service.nim:1]: ENS module has been totally refactored in master, I would suggest to take that one, this is totally different now as we interact directly with status-go

**Review by alaibe — CHANGES_REQUESTED** [review]

File: src/app_service/service/ens/service.nim (hunk 1 above)
Comment (saledjenic) [reply src/app_service/service/ens/service.nim:1]: uhhh didn't know that... will take a look soon. Thank you.

**Comment by anastasiyaig** [comment]
@saledjenic 

- [ ] once I add a ENS name, it should be selected as Primary username (currently it is none)

Screenshot 2022-01-24 at 14 41 50


- [ ] clicking Primary username modal results in empty list

Screenshot 2022-01-24 at 14 47 24

**Comment by saledjenic** [comment]
@anastasiyaig issues you found ought to be fixed now, please check it again.

**Comment by anastasiyaig** [comment]
looks great @saledjenic , lets merge it to base bc and i will test it more once we fix 1x1 chats

File: src/app/modules/main/profile_section/ens_usernames/module.nim [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  var link = network.etherscanLink.replace("/address", "/tx")
+  self.view.load(link, signingPhrase)
+
+method isLoaded*(self: Module): bool =
+  return self.moduleLoaded
+
+method viewDidLoad*(self: Module) =
+  self.fetchGasPrice()
+  # add registered ens usernames
+  let registeredEnsUsernames = self.controller.getAllMyEnsUsernames(false)
```
Comment (jrainville) [inline src/app/modules/main/profile_section/ens_usernames/module.nim:58]: Can you put the param name for clarity in from of `false`

File: src/app_service/service/ens/service.nim [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+      if success:
+        var ensUsername = self.formatUsername(username, true)
+        self.transactionService.trackPendingTransaction(response, address, $sntContract.address, 
+        $PendingTransactionTypeDto.RegisterENS, ensUsername)
+        self.pendingEnsUsernames.incl(ensUsername)
+    
+    except RpcException as e:
+      error "error occurred", methodName="registerEns"
+
+  proc getSNTBalance*(self: Service): string =
```
Comment (jrainville) [inline src/app_service/service/ens/service.nim:455]: This function can probably be moved to the wallet or balance service, but it's not a big deal

File: src/app_service/service/stickers/service.nim [hunk 4]
```diff
@@ -126,7 +126,7 @@ QtObject:
       buyToken = BuyToken(packId: packId, address: address, price: price)
       buyTxAbiEncoded = stickerMktContract.methods["buyToken"].encodeAbi(buyToken)
     approveAndCall = ApproveAndCall[100](to: stickerMktContract.address, value: price, data: DynamicBytes[100].fromHex(buyTxAbiEncoded))
-    self.eth_service.buildTokenTransaction(address, sntContract.address, gas, gasPrice, isEIP1559Enabled, maxPriorityFeePerGas, maxFeePerGas)
+    ens_utils.buildTokenTransaction(address, sntContract.address, gas, gasPrice, isEIP1559Enabled, maxPriorityFeePerGas, maxFeePerGas)
```
Comment (jrainville) [inline src/app_service/service/stickers/service.nim:129]: Maybe ens_utils isn't right. We might need an `eth_utils` or tx_utils`

**Review by jrainville — APPROVED** [review]
Nice job. I had just small suggestions

**Review by borismelnik — APPROVED** [review]

File: src/app_service/service/ens/service.nim (hunk 3 above)
Comment (saledjenic) [reply src/app_service/service/ens/service.nim:455]: @jrainville yes, I mentioned that in the first comment [here](https://github.com/status-im/status-desktop/pull/4490#issue-1107855700), actually when I started to work on this I expected that many of the things I refactored here were already done within wallet section (and related things) refactoring done, but seems not. For now we will keep this, and when one is going to refactor other part which will need some method from `ens` service he will move it to the final, proper place.

File: src/app_service/service/stickers/service.nim (hunk 4 above)
Comment (saledjenic) [reply src/app_service/service/stickers/service.nim:129]: @jrainville yes, I agree, we may find better place for that, but as I the same comment is for this as I added [here](https://github.com/status-im/status-desktop/pull/4490#discussion_r791061179). `buildTokenTransaction` may/will be used by multiple places, so wi will find better place for it, actually  `ens`, `wallet`, `transaction`, `eth` services will need a final touch at the end.

File: src/app/modules/main/profile_section/ens_usernames/module.nim (hunk 2 above)
Comment (saledjenic) [reply src/app/modules/main/profile_section/ens_usernames/module.nim:58]: @jrainville updated.

**Pull request merged**